terraform/projects/app-ecs-instances/instance-user-data.tpl

```diff
@@ -32,10 +32,16 @@ else
   echo "[$(date '+%H:%M:%S %d-%m-%Y')] attach-volume: /dev/$DEVICE is already formatted: $(file -s /dev/$DEVICE)"
 fi
 
-# Allow prometheus container user access to read/write/execute within container
+# Allow prometheus container user access to read/write metrics to persistent volume mount from within a container
 mkdir -p /ecs/prometheus_data
 mount /dev/$DEVICE /ecs/prometheus_data
-chmod 777 /ecs/prometheus_data
+
+
+#Create prometheus & grant access to prometheus data files. We create the group & match it to the gid
+groupadd --system --gid 65534 prometheus
+useradd --system --uid 65534 --gid 65534
+chown prometheus:prometheus /ecs/prometheus_data
```

Comment:
`prometheus:prometheus` refers to user with name `prometheus` and group with name `prometheus` if I understand this correctly. Do we have a user with name `prometheus`? We set up the group but I don't know if we have the user?

Reply:
Ok I took the user out for some strange reason, however I have put the user back in and all should be well :)